tests/tests/custom_authorities_operations_tests.cpp

```diff
@@ -0,0 +1,428 @@
+      
+      update_account(dan);
+      
+      {
+         auto authorities = db.get_custom_authorities_by_account(dan.id);
+         BOOST_REQUIRE_EQUAL(2, authorities.size());
+         
+         auto auth = authorities[0];
+         BOOST_CHECK(dan.id == auth.account);
+         BOOST_CHECK(operation_type_id_from_operation_type<custom_authority_create_operation>::value == auth.operation_type.value);
+         
+         BOOST_REQUIRE_EQUAL(0, auth.restrictions.size());
+         
+         //authority should be disabled
+         BOOST_CHECK(!auth.enabled);
+         
+         auth = authorities[1];
+         BOOST_CHECK(dan.id == auth.account);
+         BOOST_CHECK(operation_type_id_from_operation_type<account_update_operation>::value == auth.operation_type.value);
+         
+         BOOST_REQUIRE_EQUAL(0, auth.restrictions.size());
+         
+         //authority should be disabled
+         BOOST_CHECK(!auth.enabled);
+      }
+      
+      {
+         auto authorities = db.get_custom_authorities_by_account(sam.id);
+         BOOST_REQUIRE_EQUAL(2, authorities.size());
+         
+         auto auth = authorities[0];
+         BOOST_CHECK(sam.id == auth.account);
+         BOOST_CHECK(operation_type_id_from_operation_type<custom_authority_create_operation>::value == auth.operation_type.value);
+         BOOST_REQUIRE_EQUAL(0, auth.restrictions.size());
+         BOOST_CHECK(auth.enabled);
+         
+         auth = authorities[1];
+         BOOST_CHECK(sam.id == auth.account);
+         BOOST_CHECK(operation_type_id_from_operation_type<account_update_operation>::value == auth.operation_type.value);
+         BOOST_REQUIRE_EQUAL(0, auth.restrictions.size());
+         BOOST_CHECK(auth.enabled);
+      }
+      
+   } catch (fc::exception &e) {
+      edump((e.to_detail_string()));
+      throw;
+   }
+}
+
+BOOST_AUTO_TEST_CASE(transaction_passes_without_authorities_installed) {
+   try {
+      auto dan = create_account("dan");
+      
+      custom_authority_create_operation op;
+      op.account = dan.id;
+      op.enabled = true;
+      op.valid_from = time_point_sec(1);
+      op.valid_to = time_point_sec(2);
+      op.operation_type = operation_type_id_from_operation_type<transfer_operation>::value;
+      
+      trx.operations.push_back(op);
+      trx.validate();
+      
+      BOOST_CHECK_NO_THROW(db.push_transaction(trx, ~0));
+
+   } catch (fc::exception &e) {
+      edump((e.to_detail_string()));
+      throw;
+   }
+}
+
+BOOST_AUTO_TEST_CASE(transaction_fails_with_authorities_installed) {
+   try {
+      auto dan = create_account("dan");
+      
+      custom_authority_create_operation op;
+      op.account = dan.id;
+      op.enabled = true;
+      op.valid_from = time_point_sec(1); //validation will fail because of not valid interval
+      op.valid_to = time_point_sec(2);
+      op.operation_type = operation_type_id_from_operation_type<custom_authority_delete_operation>::value;
```

Comment:
This is a custom active that allows execution of `custom_authority_delete_operation` correct?

Maybe use a more meaningful operation, like `transfer` or `account_update`, or even better `limit_order`

Reply:
Yes, you are right. Will do. Thank you.

Reply:
Done